Flux diagram/cte_replace_labels_to_smiles v2.ipynb

```diff
@@ -0,0 +1,495 @@
+    "    i_ign = states(species).Y.argmax()\n",
+    "    return states.t[i_ign]\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "0a89dc96",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def shock_tube_simulation(estimatedIgnitionDelayTime,cantera_annottated_path,T,P,X):\n",
+    "    # Define the reactor temperature and pressure: Surjean 2013 Shock Tube first experiment out of 8\n",
+    "    reactorTemperature = T  # Kelvin\n",
+    "    reactorPressure = P  # Pascals\n",
+    "\n",
+    "    # Load the real gas mechanism:\n",
+    "    gas = ct.Solution(cantera_annottated_path)\n",
+    "    # Set the state of the gas object:\n",
+    "    gas.TP = reactorTemperature, reactorPressure\n",
+    "\n",
+    "    ##########Edit the Species labels by their SMILES except of Ar+Ne elements ##\n",
+    "    # Define the fuel, oxidizer and set the stoichiometry:\n",
+    "    #gas.X = {'CC1=CC=C(C)O1': 0.01, '[O][O]': 0.075, 'Ar': 0.915}\n",
+    "    gas.X=X\n",
+    "    #############################################################################\n",
+    "\n",
+    "    # Create a reactor object and add it to a reactor network\n",
+    "    # In this example, this will be the only reactor in the network\n",
+    "    r = ct.Reactor(contents=gas)\n",
+    "    reactorNetwork = ct.ReactorNet([r])\n",
+    "    timeHistory = ct.SolutionArray(gas, extra=['t'])\n",
+    "    # Tic\n",
+    "    t0 = time.time()\n",
+    "\n",
+    "    t = 0\n",
+    "\n",
+    "    counter = 1\n",
+    "    while t < estimatedIgnitionDelayTime:\n",
+    "        t = reactorNetwork.step()\n",
+    "        if counter % 20 == 0:\n",
+    "            # We will save only every 20th value. Otherwise, this takes too long\n",
+    "            # Note that the species concentrations are mass fractions\n",
+    "            timeHistory.append(r.thermo.state, t=t)\n",
+    "        counter += 1\n",
+    "\n",
+    "    # Toc\n",
+    "    t1 = time.time()\n",
+    "    time_computed=t1-t0\n",
+    "    return timeHistory,gas\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "3d9e7f15",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_dot_and_png_files_in_t_dir(flux_diagram_folder_path,t,tau,gas):\n",
+    "    \"\"\"\n",
+    "    This func creates a new folder to each t time interval. Inside that folder, 3 files are created:\n",
+    "    dot, modified_dot, and png file (the flux diagram itself)\n",
+    "    It's importatnt to note that the fluc diagram follows C element --> can be changed\n",
+    "    \"\"\"\n",
+    "    time=str(np.format_float_scientific(tau*t, unique=False, precision=4))#scientific notation\n",
+    "    \n",
+    "    # Define the element to follow in the reaction path diagram:\n",
+    "    element = 'C'\n",
+    "\n",
+    "    if t==1:\n",
+    "        t_path=flux_diagram_folder_path+\"/tau\"\n",
```

Comment:
Could you use `os.path.join()` here as well, and anywhere else when combining paths?